curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md

```diff
@@ -1,49 +1,53 @@
 ---
 id: 587d7b88367417b2b2512b45
-title: Use Higher-Order Functions map, filter, or reduce to Solve a Complex Problem 
+title: Use Higher-Order Functions map, filter, or reduce to Solve a Complex Problem
 challengeType: 1
 forumTopicId: 301311
 ---
 
 ## Description
+
 <section id='description'>
 Now that you have worked through a few challenges using higher-order functions like <code>map()</code>, <code>filter()</code>, and <code>reduce()</code>, you now get to apply them to solve a more complex challenge.
 </section>
 
 ## Instructions
+
 <section id='instructions'>
 We have defined a function named <code>squareList</code>.  You need to complete the code for the <code>squareList</code> function using any combination of <code>map()</code>, <code>filter()</code>, and <code>reduce()</code> so that it returns a new array containing only the square of <em>only</em> the positive integers (decimal numbers are not integers) when an array of real numbers is passed to it.  An example of an array containing only real numbers is <code>[-3, 4.8, 5, 3, -3.2]</code>.
 <strong>Note:</strong> Your function should not use any kind of <code>for</code> or <code>while</code> loops or the <code>forEach()</code> function.
 </section>
 
 ## Tests
+
 <section id='tests'>
 
 ```yml
 tests:
   - text: <code>squareList</code> should be a <code>function</code>.
     testString: assert.typeOf(squareList, 'function'), '<code>squareList</code> should be a <code>function</code>';
-  - text: for or while loops or forEach should not be used.
+  - text: <code>for</code> or <code>while</code> loops or <code>forEach</code> should not be used.
     testString: assert(!__helpers.removeJSComments(code).match(/for|while|forEach/g));
   - text: <code>map</code>, <code>filter</code>, or <code>reduce</code> should be used.
-    testString: assert(__helpers.removeJSComments(code).match(/\.(map|filter|reduce)\s*\(/g));
+    testString: assert(__helpers.removeWhiteSpace(__helpers.removeJSComments(code)).match(/\.(map|filter|reduce)\(/g));
```

Comment:
If you are just looking for the words `map`, `filter`, or `reduce`, I do not see the reason to remove the whitespace.